openstack/sre/templates/prometheus-alerts.yaml

```diff
@@ -13,24 +13,65 @@ spec:
     rules:
-    - alert: ApiAvailabilityWarning
+    # quick availability check - target updates
+    - alert: ApiAvailabilityWarning1Hour
       expr: (1 - global:api_errors_per_request_sli:ratio_rate1h) *100 <= on (ingress,region) global:api_availability_warning_slo:percent
       labels:
         context: availability
         service: "{{`{{$labels.ingress}}`}}"
         severity: warning
         tier: sre
+        playbook: 'docs/devops/alert/sre-alerts.html'
       annotations:
-        summary: "An API fell below availability warning SLO in the last hour"
-        description: "{{`{{$labels.ingress}}`}} fell below availability warning SLO in the last hour."
+        summary: "API availability warning SLO hit: 1h"
+        description: "{{`{{$labels.ingress}}`}} fell below availability warning SLO in the last hour (default:  *99.8%*)."
 
-    - alert: ApiAvailabilityCritical
+    - alert: ApiAvailabilityCritical1Hour
       expr: (1 - global:api_errors_per_request_sli:ratio_rate1h) *100 <= on (ingress,region) global:api_availability_critical_slo:percent
       labels:
         context: availability
         service: "{{`{{$labels.ingress}}`}}"
         severity: warning
         tier: sre
+        playbook: 'docs/devops/alert/sre-alerts.html'
       annotations:
-        summary: "An api fell below availability critical SLO in the last hour"
-        description: "{{`{{$labels.ingress}}`}} fell below availability critical SLO in the last hour."
+        summary: "API availability critical SLO hit: 1h"
+        description: "{{`{{$labels.ingress}}`}} fell below availability critical SLO in the last hour (default:  *99.6%*)."
+
+    # rolling 7 days
+    - alert: ApiAvailabilityWarningLast7days
+      expr: (1 - global:api_errors_per_request_sli:ratio_rate7d) *100 <= on (ingress,region) global:api_availability_critical_slo:percent
+      labels:
+        context: availability
+        service: "{{`{{$labels.ingress}}`}}"
+        severity: info
+        tier: os
+        playbook: 'docs/devops/alert/sre-alerts.html'
+      annotations:
+        summary: "API availability warning SLO hit: 7d"
+        description: "{{`{{$labels.ingress}}`}} fell below weekly availability warning SLO (default:  *99.8%*)."
+
+    - alert: ApiAvailabilityCriticalLast7days
+      expr: (1 - global:api_errors_per_request_sli:ratio_rate7d) *100 <= on (ingress,region) global:api_availability_critical_slo:percent
+      labels:
+        context: availability
+        service: "{{`{{$labels.ingress}}`}}"
+        severity: info
+        tier: os
+        playbook: 'docs/devops/alert/sre-alerts.html'
+      annotations:
+        summary: "API availability critical SLO hit: 7d"
+        description: "{{`{{$labels.ingress}}`}} ffell below weekly availability warning SLO (default:  *99.6%*)."
+
+    # meta alert
+    - alert: SRE-MetricAbesent
+      expr: absent(global:api_errors_per_request_sli:ratio_rate1h) or absent(global:api_availability_critical_slo:percent)
```

Comment:
How do we define that this alert prohibits those other alerts from above from firing? Or do the ones above not fire at all if one of the metrics is missing?

What about the `global:api_errors_per_request_sli:ratio_rate7d` we use above? Don't we have to check for its absence, too?

Reply:
Yes @ "Or do the ones above not fire at all if one of the metrics is missing"

YNeos @  "Don't we have to check for its absence, too" : 
- yes part: to be on the safe side we should check, too
- no part: normally I just choose one metric as sample for the concept (it's a ~8o% approach of risk coverage)

Reply:
ok. then let's keep it that way.